packages/botonic-react/src/util/webchat.js

```diff
@@ -0,0 +1,151 @@
+  defaultValue = undefined
+) => {
+  for (const [k, v] of Object.entries(WEBCHAT.CUSTOM_PROPERTIES)) {
+    if (v == property) {
+      const nestedProperty = getProperty(theme, v)
+      if (nestedProperty !== undefined) return nestedProperty
+      const plainProperty = getProperty(theme, k)
+      if (plainProperty !== undefined) return plainProperty
+      return defaultValue
+    }
+  }
+  return undefined
+}
+
+export class ButtonsDisabler {
+  static constructBrowserProps(props) {
+    const disabledProps = { disabled: props.disabled }
+    if (props.autodisable !== undefined)
+      disabledProps.autodisable = strToBool(props.autodisable)
+    if (props.disabledstyle !== undefined)
+      disabledProps.disabledstyle = props.disabledstyle
+    return disabledProps
+  }
+  static constructNodeProps(props) {
+    const disabledProps = {}
+    if (props.autodisable !== undefined)
+      disabledProps.autodisable = String(props.autodisable)
+    if (props.disabledstyle !== undefined)
+      disabledProps.disabledstyle = JSON.stringify(props.disabledstyle)
+    return disabledProps
+  }
+
+  static withDisabledProps(props) {
+    return {
+      ...{ disabled: props.disabled },
+      ...{ autodisable: props.autodisable },
+      ...{ disabledstyle: props.disabledstyle },
+    }
+  }
+
+  static resolveDisabling(theme, props) {
+    const getThemeProperty = _getThemeProperty(theme)
+    const autoDisable =
+      props.autodisable !== undefined
+        ? props.autodisable
+        : getThemeProperty(
+            WEBCHAT.CUSTOM_PROPERTIES.buttonAutoDisable,
+            WEBCHAT.DEFAULTS.BUTTON_AUTO_DISABLE
+          )
+    const disabledStyle =
+      props.disabledstyle !== undefined
+        ? props.disabledstyle
+        : getThemeProperty(
+            WEBCHAT.CUSTOM_PROPERTIES.buttonDisabledStyle,
+            WEBCHAT.DEFAULTS.BUTTON_DISABLED_STYLE
+          )
+    return { autoDisable, disabledStyle }
+  }
+
+  static updateChildrenButtons(children, additionalProps = undefined) {
+    return deepMapWithIndex(children, n => {
+      if (n.type === Button) return this.updateButtons(n, additionalProps)
+      if (n.props && n.props.children) {
+        return {
+          ...n,
+          ...{
+            props: {
+              ...n.props,
+              ...{ children: deepMapWithIndex(n.props.children, n => n) },
+            },
+          },
+        }
+      }
+      return n
+    })
+  }
+
+  static updateButtons(node, additionalProps) {
+    if (!additionalProps) additionalProps = {}
+    else {
+      additionalProps = {
+        disabled:
+          node.props.disabled === true
+            ? node.props.disabled
+            : additionalProps.disabled,
+        setDisabled: additionalProps.setDisabled,
+        parentId: additionalProps.parentId,
+      }
+    }
+    return {
+      ...node,
+      props: {
+        ...node.props,
+        autodisable: node.props.autodisable,
+        disabledstyle: node.props.disabledstyle,
```

Comment:
Are these 2 lines needed? Since both properties (`autodisable` and `disabledstyle`) are present in `node.props`, it should be enought with the previous line (`...node.props`), right?

Reply:
sure, applied